src/App.tsx

```diff
@@ -9,52 +10,192 @@
+            <div className="tile is-child box is-success">
+              <div className="block">
+                {isLoadingUsers || isLoadingPosts ? (
+                  <Loader />
+                ) : (
+                  <UserSelector
+                    selectedUser={selectedUser}
+                    users={users}
+                    onSelect={setSelectedUser}
+                  />
+                )}
               </div>
 
-              <div className="notification is-warning" data-cy="NoPostsYet">
-                No posts yet
+              <div className="block" data-cy="MainContent">
+                {selectedUser && !isLoadingPosts && !error ? (
+                  posts.length > 0 ? (
+                    <PostsList
+                      posts={posts}
+                      selectedPost={selectedPost}
+                      onOpen={setSelectedPost}
+                      onClose={() => {
+                        setSelectedPost(null);
+                      }}
+                    />
+                  ) : (
+                    <div
+                      className="notification is-warning"
+                      data-cy="NoPostsYet"
+                    >
+                      No posts yet
+                    </div>
+                  )
+                ) : (
+                  <p data-cy="NoSelectedUser">No user selected</p>
+                )}
               </div>
 
-              <PostsList />
+              {error && (
+                <div
+                  className="notification is-danger"
+                  data-cy="PostsLoadingError"
+                >
+                  {error}
+                </div>
+              )}
             </div>
           </div>
-        </div>
-
-        <div
-          data-cy="Sidebar"
-          className={classNames(
-            'tile',
-            'is-parent',
-            'is-8-desktop',
-            'Sidebar',
-            'Sidebar--open',
+          {selectedPost && (
+            <div
+              data-cy="Sidebar"
+              className={classNames(
+                'tile',
+                'is-parent',
+                'is-8-desktop',
+                'Sidebar',
+                'Sidebar--open',
+              )}
+            >
+              <div className="tile is-child box is-success ">
+                <PostDetails
+                  post={selectedPost}
+                  error={error}
+                  comments={comments}
+                  formOpen={setIsFormOpen}
+                  isFormOpen={isFormOpen}
+                  deleteComment={handleDeleteComment}
+                  addComment={handleAddComment}
+                  isLoadingComments={isLoadingComments}
+                />
+              </div>
+            </div>
           )}
-        >
-          <div className="tile is-child box is-success ">
-            <PostDetails />
-          </div>
         </div>
       </div>
-    </div>
-  </main>
-);
+    </main>
+  );
+};
```

Comment:
Consider defining `propTypes` for your components to ensure proper type-checking and that components receive the correct props. This is crucial for maintaining code quality and avoiding runtime errors.